src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -72,8 +72,29 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
 
             try
             {
+                ExportJobConfiguration exportJobConfiguration = _exportJobConfiguration;
+
+                string connectionHash = Microsoft.Health.Core.Extensions.StringExtensions.ComputeHash(_exportJobConfiguration.StorageAccountConnection);
+
+                if (string.IsNullOrEmpty(exportJobRecord.StorageAccountUri) && string.IsNullOrEmpty(_exportJobConfiguration.StorageAccountUri))
+                {
+                    if (!string.Equals(exportJobRecord.StorageAccountConnectionHash, connectionHash, StringComparison.Ordinal))
```

Comment:
In this scenario we don't attempt to connect to the old "StorageAccountConnection" whereas in the below scenario we attempt to connect to the old "StorageAccountUri". It might be better to have the same behavior for both cases.

Reply:
Because we are only passing a hash of the connection string we have no way of connecting to the old storage account if the job used the connection string. I can change the behavior to be the same with the uri, but we have the ability to try to finish the job with the uri.

Reply:
Aah. Agreed.